From 45e6e9db9f49b93f546d0afda8531a8d4b783f28 Mon Sep 17 00:00:00 2001
From: Michel Lespinasse <michel@lespinasse.org>
Date: Mon, 1 Nov 2021 10:39:08 -0700
Subject: [PATCH 01/34] scripts/package/builddeb: add UPLOAD_DIR and
 NO_HEADER_PACKAGES

If UPLOAD_DIR is set, create packages in that directory rather than ..

If NO_HEADER_PACKAGES is set, skip creating the linux-headers and
linux-libc-dev packages.
---
 scripts/package/builddeb | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index 91a502bb9..b83a4a09a 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -50,7 +50,7 @@ create_package() {
 
 	# Create the package
 	dpkg-gencontrol -p$pname -P"$pdir"
-	dpkg-deb $dpkg_deb_opts ${KDEB_COMPRESS:+-Z$KDEB_COMPRESS} --build "$pdir" ..
+	dpkg-deb $dpkg_deb_opts ${KDEB_COMPRESS:+-Z$KDEB_COMPRESS} --build "$pdir" "${UPLOAD_DIR:-..}"
 }
 
 deploy_kernel_headers () {
@@ -208,7 +208,7 @@ EOF
 	chmod 755 "$tmpdir/DEBIAN/$script"
 done
 
-if [ "$ARCH" != "um" ]; then
+if [ "$ARCH" != "um" -a -z "$NO_HEADER_PACKAGES" ]; then
 	if is_enabled CONFIG_MODULES; then
 		deploy_kernel_headers debian/linux-headers
 		create_package linux-headers-$version debian/linux-headers
-- 
2.33.1.711.g9d530dc002

